tornado-api/src/main/java/uk/ac/manchester/tornado/api/plantype/ExecutionPlanType.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2024, APT Group, Department of Computer Science,
+ * The University of Manchester.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.manchester.tornado.api.plantype;
+
+import uk.ac.manchester.tornado.api.TornadoExecutionPlan;
+
+public sealed class ExecutionPlanType extends TornadoExecutionPlan //
```

Comment:
Do you think it makes sense to make the ExecutionPlan types records instead?

Reply:
ExecutionPlanType cotains mutable fields and it extends another class, so we can't.